onnxruntime/core/providers/xnnpack/xnnpack_execution_provider.cc

```diff
@@ -87,14 +88,38 @@ std::unique_ptr<KernelRegistry> RegisterKernels() {
 using namespace xnnpack;
 
 XnnpackExecutionProvider::XnnpackExecutionProvider(const XnnpackExecutionProviderInfo& info)
-    : IExecutionProvider{kXnnpackExecutionProvider, true} {
+    : IExecutionProvider{kXnnpackExecutionProvider, true}, info_(info) {
   if (info.xnn_thread_pool_size > 1) {
     // pthreadpool is independent of ort-threadpoool, so we have to disable cpu spinning for ort-threadpool.
     // otherwise, the pthreadpool will be starved and harm performance a lot.
     xnnpack_thread_pool_ = pthreadpool_create(static_cast<size_t>(info.xnn_thread_pool_size));
   }
 }
 
+common::Status XnnpackExecutionProvider::CheckSessionOptions(const SessionOptions& so) {
+  if (so.execution_mode != ExecutionMode::ORT_SEQUENTIAL) {
+    return ORT_MAKE_STATUS(ONNXRUNTIME, FAIL,
+                           "Parallel execution mode does not support the Xnnpack Execution Provider. "
+                           "Please set the execution mode as sequential for this "
+                           "session since it uses the Xnnpack Execution Provider.");
+  }
```

Comment:
What's the reason for this? For CUDA it's due to how the CUDA streams work. What about the xnnpack kernel execution would not work with the parallel executor?

Reply:
For all xnn_setup_xxxx like function, it will change xnn_operator_t.

Reply:
We prevent concurrent calls to Run to avoid an individual node being executed concurrently for that reason. However, the parallel executor is not going to execute an individual node concurrently - it will potentially execute different nodes concurrently which would all be separate instances of the kernel.

Reply:
Got it.